New: Support component folders and include them in output interfaces

svrin opened the pull request:
Hi,

this PR adds the possibility to include subfolders of the component folders and output them as interfaces.

aperron commented on src/importer.ts:
```diff
@@ -76,50 +76,68 @@ export const findFiles = (dir: string, ext: RegExp = /.settings.json$/) =>
-            console.warn(`Already have model '${strapiModel.info.name}' => skip ${results[sameNameIndex]._filename} use ${strapiModel._filename}`)
-            results[sameNameIndex] = strapiModel;
+            console.warn(`Already have component '${strapiModel.info.name}' => skip ${components[sameNameIndex]._filename} use ${strapiModel._filename}`)
+            components[sameNameIndex] = strapiModel;
```
```suggestion
            components[sameNameIndex] = {
              ...strapiModel,
              info: {
                ...strapiModel.info,
                name: `${componentCollectionName}.${componentModelName}`
              }
            };
```

aperron reviewed (changes requested):
sorry for the time taken to review your PR.

I tested with a component and interfaceName of this component has a dot (Enum of this component too).  Example:
```typescript
export interface Xorder.payment {
  id: string;
  mode: EnumXorder.paymentmode;
  count: number;
  url?: string;
  status: EnumXorder.paymentstatus;
}
```

Could you add a test with a component ?
For example,
* simply add `    inputGroup: 'src/test/components',` in `src/test/config.js`
*  add *.json in  `src/test/components/*/models/`
* run `yarn test`

svrin commented:
> sorry for the time taken to review your PR.

no worries, will add test next week, thx for looking over it!

svrin commented:
@aperron: I Added three components for testing, the . in the name is due the overwritten `toInterfaceName` in the config.js.

But I don't know how to handle this well, I added now a warning if an overwritten `toInterfaceName` returns a name with a ".", or should we separate toInterfaceName into modelToInterfaceName and componentToInterfaceName for overwriting?

aperron commented:
Hello,

I review your PR with your last changes. I think that separating the import of the components and the models adds a little too much complexity. So i add a field _isComponent on IStrapiModel during import.
I take your test data and some of your modifications.
I just need more time to do non regression tests. For that I added the possibility to write several tests with different configurations.
If you want to watch the latest version on the master branch (not yet release), I would be happy to have your feedback.

aperron commented:
Your modification are integrated in v2.0.9. Thanks.  
Feel free to create an other PR if not all are there.

svrin commented:
I found one little bug: If you have named a component same as a content-type there will be currently some struggle. 

I will provide a test case and an error report some days later, but for now it works, thx a lot.
